analyzers/src/SonarAnalyzer.CSharp/Rules/PrivateStaticMethodUsedOnlyByNestedClass.cs

```diff
@@ -102,25 +102,18 @@ private static bool IsPrivateAndStatic(MethodDeclarationSyntax method, TypeDecla
 
     private static TypeDeclarationSyntax LowestCommonAncestorOrSelf(IEnumerable<TypeDeclarationSyntax> declaredTypes)
     {
-        var typeHierarchyFromTopToBottom = declaredTypes.Select(PathFromTop);
+        var typeHierarchyFromTopToBottom = declaredTypes.Select(PathFromTop).ToArray();
         var minPathLength = typeHierarchyFromTopToBottom.Select(x => x.Length).Min();
         var firstPath = typeHierarchyFromTopToBottom.First();
+        var levels = Enumerable.Range(0, minPathLength);
+        var firstDifferingLevel = levels.FirstOrDefault(NotEveryNodeIsTheSameOnLevel);
+        int levelOfLowestCommonAncestor = firstDifferingLevel == default
+            ? minPathLength - 1
+            : firstDifferingLevel - 1;
```

Comment:
having `-1` makes it a bit difficult to follow the logic...

there's another implicit negation here: `Differ` rather then `Common` in the name of the method: `LowestCommonAncestorOrSelf`

So the method is named `Common`, but the logic inside is looking for `Different`